### PR TITLE
feat: observe team app lock config (WPB-4476)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4476" title="WPB-4476" target="_blank">TaskWPB-4476</a>  Enforcing App Lock Based on Team Settings
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

In this PR, I updated mainly `ObserveAppLockConfigUseCase` so it will observe  team app lock config as well and return the valid status for the app

Next PR will be for displaying SetAppLockScreen when it got enabled by the team
 
Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1567](https://10.10.124.134/job/AR-build-pipeline/1567/) **failed**.

**Comment by AndroidBob** [comment]
Build [1571](https://10.10.124.134/job/AR-build-pipeline/1571/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7945140)(7945140)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7945140)(7945140)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt [hunk 1]
```diff
@@ -391,6 +391,14 @@ class UseCaseModule {
...
         coreLogic.getSessionScope(currentAccount).markGuestLinkFeatureFlagAsNotChanged
 
+//    @ViewModelScoped
+//    @Provides
+//    fun provideAppLockTeamFeatureConfigObserver(
+//        @KaliumCoreLogic coreLogic: CoreLogic,
+//        @CurrentAccount currentAccount: UserId
+//    ): AppLockTeamFeatureConfigObserver =
+//        coreLogic.getSessionScope(currentAccount).appLockTeamFeatureConfigObserver
+
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt]: 🤔 would this be waiting for Kalium's PR to be merged?

**Review by alexandreferris — COMMENTED** [review]
great work! just a tiny question on the commented code 🤔

File: app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt]: Not needed in this PR, I will remove it

**Comment by AndroidBob** [comment]
Build [1600](https://10.10.124.134/job/AR-build-pipeline/1600/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7978315)(7978315)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7978315)(7978315)-staging-compat.apk)

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1611](https://10.10.124.134/job/AR-build-pipeline/1611/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7987022)(7987022)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7987022)(7987022)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [1636](https://10.10.124.134/job/AR-build-pipeline/1636/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7996024)(7996024)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7996024)(7996024)-staging-compat.apk)

**Review by borichellow — APPROVED** [review]

**Pull request merged**